examples/tutorials/analysis-3d/event_sampling_nrg_depend_models.py

```diff
@@ -0,0 +1,384 @@
+
+irfs = load_irf_dict_from_file(path / irf_filename)
+location = observatory_locations["cta_south"]
+
+observation = Observation.create(
+    obs_id=1001,
+    pointing=pointing,
+    livetime=livetime,
+    irfs=irfs,
+    location=location,
+)
+
+######################################################################
+
+energy_axis = MapAxis.from_energy_bounds("0.2 TeV", "100 TeV", nbin=5, per_decade=True)
+energy_axis_true = MapAxis.from_energy_bounds(
+    "0.05 TeV", "150 TeV", nbin=10, per_decade=True, name="energy_true"
+)
+migra_axis = MapAxis.from_bounds(0.5, 2, nbin=150, node_type="edges", name="migra")
+
+geom = WcsGeom.create(
+    skydir=pointing,
+    width=(2, 2),
+    binsz=0.02,
+    frame="icrs",
+    axes=[energy_axis],
+)
+
+######################################################################
+
+empty = MapDataset.create(
+    geom,
+    energy_axis_true=energy_axis_true,
+    migra_axis=migra_axis,
+    name="my-dataset",
+)
+maker = MapDatasetMaker(selection=["exposure", "background", "psf", "edisp"])
+dataset = maker.run(empty, observation)
+
+dataset.models = models
+
+print(dataset.models)
+
+
+######################################################################
+# Let's simulate the model
+# ~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Initialize and run the `MapDatasetEventSampler` class. We also define
+# the `oversample_energy_factor` arguments: this should be carefully
+# considered by the user, as a higher `oversample_energy_factor` gives
+# a more precise source flux estimate, at the expense of computational
+# time. Here we adopt an `oversample_energy_factor` of 10:
+#
+
+sampler = MapDatasetEventSampler(random_state=0, oversample_energy_factor=10)
+events = sampler.run(dataset, observation)
+
+######################################################################
+# Let's inspect the simulated events in the source region:
+#
+
+src_position = SkyCoord(100.0, 30.0, frame="icrs", unit="deg")
+
+on_region_radius = Angle("0.15 deg")
+on_region = CircleSkyRegion(center=src_position, radius=on_region_radius)
+
+src_events = events.select_region(on_region)
+
+src_events.peek()
+
+
+######################################################################
+# Fit the simulated data
+# ~~~~~~~~~~~~~~~~~~~~~~
+#
+# We fit the events in one time bin, adopting a model given by a
+# powerlaw spectral model, with no temporal information and point-like
+# morphology. We fix the source coordinates during the fit for simplicity:
+#
+
+spec = PowerLawSpectralModel(
+    index=2, amplitude=5e-11 * u.cm**-2 * u.s**-1 * u.TeV**-1, reference="1 TeV"
+)
+
+spatial_model = PointSpatialModel(lon_0="100 deg", lat_0="30 deg", frame="icrs")
+
+model = SkyModel(
+    spatial_model=spatial_model,
+    spectral_model=spec,
+    name="fit_source",
+)
+
+model.spatial_model.parameters[0].frozen = True
+model.spatial_model.parameters[1].frozen = True
+
+bkg_model = FoVBackgroundModel(dataset_name="my-dataset")
+
+model_fit = [model, bkg_model]
```

Comment:
Is it a good idea to fit the background within 100 sec bins? @registerrier ?

Reply:
This is certainly not a good estimate of the bkg, in general. However, in this case we are assuming to know it very well (it is read from the IRF) and the bkg normalization is compatible with 1. So, it is not the best approach but does not introduce a large bias.

Reply:
I dont know what would be the best to show in a tutorial. The background model should be shared between all the split bins, so they are not really independent... In practice, what might be good is to first fit the background on the observation, masking the source. Then split the observation into smaller bins, and keep the parameters of the background model fixed to the previously obtained value. This would be too complex (and unnecessary) to show in this tutorial.

I am just a bit afraid that people trying to fit the background model on very small time scales may end up with nonsense results

Reply:
Maybe, rather than get into all this, we can use the `LightCurveEstimator` for 3 energy bins which can show the spectral evolution.

Reply:
yes, probably the approach you propose to estimate the BKG is too much for the aims of the tutorial. What about if we just write a warning to the reader that this is only a very simplified example and that the bkg estimate might be misleading using the method we show?